lib/ansible/modules/cloud/google/gce_snapshot.py

```diff
@@ -192,9 +192,11 @@ def main():
 
     for instance_disk in instance_disks:
         disk_snapshot_name = snapshot_name
-        device_name = instance_disk['deviceName']
+        disk_info = gce._get_components_from_path(instance_disk['source'])
+        device_name = disk_info['name']
```

Comment:
The example in your PR made it look like the name listed under source is different than the deviceName. Is this true?

Reply:
When instance is created from Google UI, the device name is same as part of source name.
Whereas, when instance is created using gce module, the device name is different than that of source name.